### ISSUE TITLE
Feature request: Latest version online

### ISSUE THREAD

**Issue by Defcon0** [contao-community-alliance/composer-client#291]
Hello folks,

at first thanks for the great job done with this plugin!

I just have a little feature request. The default setting installing a composer extension in the composer client is something like ">=1.2.4.0,<1.3-dev". When a new version is created, meaning 1.3.x, I get no notice about that at all. Of course, 1.3.x is *not* innstalled by clicking "update packages". But a little note about that would be cool.

Especially new coworkers not familiar with composer think, they work on the current version of some package, but in fact they don't.

An additional column in the table would be awesome.

Do you think this would be possible in a future release?

Bye Defcon0

**Comment by discordier** [comment]
I don't think that this is really possible.
This would require to run a solving pass (in order to know what the latest installable version is).

Simply showing which version is the bleeding edge is not sufficient, as that one most likely will violate either your specified constraint or the constraint of any other dependency.

Second, we are pretty short on manpower, we are working on several other projects and this client was started only because of our own needs. That is is of benefit to the whole community is a very nice coincidence.
Therefore we will not implement it on our own but you are very welcome to file a pull request.

Sidenote: Did you notice the test run button in the backend? that one will show you in the console log what updates are available and will get installed by clicking on the upgrade button.

**Comment by fritzmg** [comment]
imho it would suffice to simply show the latest version and not run a solving pass.

**Comment by discordier** [comment]
This will, for sure, confuse the average user as they will see many new versions available which will simply fail to install.
In the end, the whole concept of the constraints is to ensure validity, you now suggest that we show versions that will most likely fail to install.

**Comment by fritzmg** [comment]
Isn't it already confusing for the average user to not be able to see whether there are updates for a package or to use the dry run to achieve that? ;) 

Especially nowadays, since the dry run generates messages about the core bundles.

**Comment by discordier** [comment]
You are right about the core bundles, we did not find a fix for this one yet.
However, I don't think that adding more confusion is the cure to confusion. :smile_cat:

**Comment by fritzmg** [comment]
True.

Hm, wouldn't it be possible to analyse the outcome of the dry run, which packages would have been updated, save that as a json file and display the information in the backend? And then when you run an update, it gets deleted again

**Comment by discordier** [comment]
Not easily from outside of composer.phar.

I implemented a workaround in the contao manager for exactly this (See https://github.com/tenside/core/commit/f2f13784d8d61ebd4e08de7e6213c99e494baa7e for details).
However, this workaround expects to be run within composer, maybe you want to play around with this idea how it could be integrated in the client (will most likely require another plugin).
